### ISSUE TITLE
App crashes when opening app drawer

### ISSUE THREAD

**Issue by PorygonZRocks** [OpenLauncherTeam/openlauncher#291]
#### General information

* **App Version:** 0.5.8(alpha)
* **Android Version:** 7.1.2

#### Expected result
**What is expected?** 
The app drawer should open. 

**What does happen instead?**
The app crashes (error details below).
```Build version: 0.5.8(alpha) 
Build date: 1979-11-30 00:00:00 
Current date: 2017-12-01 22:54:27 
Device: LGE LG-v410 
 
Stack trace:  
java.lang.StringIndexOutOfBoundsException: length=20; regionStart=0; regionLength=21
	at java.lang.String.substring(String.java:1931)
	at com.benny.openlauncher.core.widget.AppItemView.onDraw(AppItemView.java:196)
	at android.view.View.draw(View.java:17189)
	at android.view.View.updateDisplayListIfDirty(View.java:16171)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at android.support.v7.widget.RecyclerView.drawChild(RecyclerView.java:4607)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.draw(View.java:17192)
	at android.support.v7.widget.RecyclerView.draw(RecyclerView.java:4013)
	at android.view.View.updateDisplayListIfDirty(View.java:16171)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.updateDisplayListIfDirty(View.java:16166)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.draw(View.java:17192)
	at android.view.View.updateDisplayListIfDirty(View.java:16171)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at io.codetail.widget.RevealFrameLayout.drawChild(RevealFrameLayout.java:32)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.draw(View.java:17192)
	at android.view.View.updateDisplayListIfDirty(View.java:16171)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
	at android.view.View.updateDisplayListIfDirty(View.java:16134)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
	at android.view.View.updateDisplayListIfDirty(View.java:16134)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at android.support.v4.widget.DrawerLayout.drawChild(DrawerLayout.java:1367)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.updateDisplayListIfDirty(View.java:16166)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
	at android.view.View.updateDisplayListIfDirty(View.java:16134)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
	at android.view.View.updateDisplayListIfDirty(View.java:16134)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
	at android.view.View.updateDisplayListIfDirty(View.java:16134)
	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
	at android.view.View.updateDisplayListIfDirty(View.java:16134)
	at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:648)
	at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:654)
	at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:762)
	at android.view.ViewRootImpl.draw(ViewRootImpl.java:2800)
	at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2608)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2215)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1254)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6344)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:874)
	at android.view.Choreographer.doCallbacks(Choreographer.java:686)
	at android.view.Choreographer.doFrame(Choreographer.java:621)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:860)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6186)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:107)
```

#### Steps to reproduce
* I just open the app drawer and it crashes. No clue how somebody else could reproduce it.

**Comment by testman42** [comment]
Can confirm. Just updated OpenLauncher from F-Droid, now it crashes every time when it should open app drawer.
Plz fix, thx.

**Comment by offa** [comment]
I have the same problem, but following workaround helped to get it running:

1. Open the App settings (not the OpenLauncher settings)
1. Clean Cache and Memory
1. Restart OpenLauncher

**Comment by testman42** [comment]
Can confirm, it works.

**Comment by PorygonZRocks** [comment]
Same. I can scroll, but if I change settings, then scrolling crashes. Basically same error message, but with at least one small difference (`regionLength=27` instead of 21). New log below, in case there are more differences. 

```Build version: 0.5.8(alpha) 
Build date: 1979-11-30 00:00:00 
Current date: 2017-12-08 13:06:25 
Device: LGE LG-v410 
 
Stack trace:  
java.lang.StringIndexOutOfBoundsException: length=25; regionStart=0; regionLength=27
	at java.lang.String.substring(String.java:1931)
	at com.benny.openlauncher.core.widget.AppItemView.onDraw(AppItemView.java:196)
	at android.view.View.draw(View.java:17189)
	at android.view.View.updateDisplayListIfDirty(View.java:16171)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at android.support.v7.widget.RecyclerView.drawChild(RecyclerView.java:4607)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.draw(View.java:17192)
	at android.support.v7.widget.RecyclerView.draw(RecyclerView.java:4013)
	at android.view.View.updateDisplayListIfDirty(View.java:16171)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.updateDisplayListIfDirty(View.java:16166)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.draw(View.java:17192)
	at android.view.View.updateDisplayListIfDirty(View.java:16171)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at io.codetail.widget.RevealFrameLayout.drawChild(RevealFrameLayout.java:32)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.draw(View.java:17192)
	at android.view.View.updateDisplayListIfDirty(View.java:16171)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.updateDisplayListIfDirty(View.java:16166)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.draw(View.java:17192)
	at android.view.View.updateDisplayListIfDirty(View.java:16171)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at android.support.v4.widget.DrawerLayout.drawChild(DrawerLayout.java:1367)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.updateDisplayListIfDirty(View.java:16166)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.updateDisplayListIfDirty(View.java:16166)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.updateDisplayListIfDirty(View.java:16166)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.updateDisplayListIfDirty(View.java:16166)
	at android.view.View.draw(View.java:16955)
	at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
	at android.view.View.draw(View.java:17192)
	at com.android.internal.policy.DecorView.draw(DecorView.java:756)
	at android.view.View.updateDisplayListIfDirty(View.java:16171)
	at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:648)
	at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:654)
	at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:762)
	at android.view.ViewRootImpl.draw(ViewRootImpl.java:2800)
	at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2608)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2215)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1254)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6344)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:874)
	at android.view.Choreographer.doCallbacks(Choreographer.java:686)
	at android.view.Choreographer.doFrame(Choreographer.java:621)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:860)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6186)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:107)
```

**Comment by EmmanuelMess** [comment]
This happened after I added a widget (Google notes) and tried to open drawer. 
Motorola Moto E Gen 2 4G
Android 6.0

**Comment by PorygonZRocks** [comment]
Why was this closed?

**Comment by EmmanuelMess** [comment]
It was fixed in #297 .

**Comment by PorygonZRocks** [comment]
Ah ok, Thanks.
